15 - Update (Elaborato.java): can store Row instance in a List

antoniocaia opened the pull request:
Closes #15

antoniocaia committed:
Init (Row.java): create class to store data

Each Row instance keep the information of a single read row.

antoniocaia committed:
Update (Elaborato.java): can store Row instance in a List

Closes #15

Wabri commented on GestioneOrari/src/Elaborato/Elaborato.java:
```diff
@@ -1,8 +1,23 @@
+		rowsList = new ArrayList<>();
+	}
+	
+	public void addNewRowToRowsList(Date date, String startHour, String endHour, String order, String note) {
```
Per add si intende che sia nuovo quindi puoi anche troncare il new dal nome e anche List non è necessario di solito quando si mette un plurale a un campo si tratta di una collection.

```suggestion
	public void addRowToRows(Date date, String startHour, String endHour, String order, String note) {
```

Wabri commented on GestioneOrari/src/Elaborato/Elaborato.java:
```diff
@@ -1,8 +1,23 @@
+
+import row.Row;
+
 public class Elaborato {
```
So che l'ho fatto io, ma riesci a trovare un nome in inglese? Manteniamo la convenzione inglese.

Pensando che è un dato elaborato si immagina che sia un risultato di un procedimento quindi un dato processato, forse Processed o qualcosa di simile come nome della classe sarebbe meglio.

Wabri commented on GestioneOrari/src/Elaborato/Elaborato.java:
```diff
@@ -1,8 +1,23 @@
+import row.Row;
+
 public class Elaborato {
+	private List<Row> rowsList;
```
List non è necessario di solito quando si mette un plurale a un campo si tratta di una collection in generale, quindi ti direi di togliere List dal nome.

```suggestion
	private List<Row> rows;
```

Wabri commented on GestioneOrari/src/Elaborato/Elaborato.java:
```diff
@@ -1,8 +1,23 @@
+		rowsList.add(new Row(date, startHour, endHour, order, note));
+	}
+	
+	public List<Row> getRowsList(){
```
stesse considerazioni sul nome fatte per il metodo e il nome del campo, list non è necessaria.

Wabri commented on GestioneOrari/src/row/Row.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@AllArgsConstructor
```
Bravissimo, non conoscevo questa annotazione. Ho dovuto andare a cercarla, se riesci metti un commento con il link relativo alla documentazione ufficiale https://projectlombok.org/api/lombok/AllArgsConstructor.html

(i commenti sono sbagliati di norma, quando faremo un refactor inseriremo i javadocs e elimineremo ogni commento non necessario)

antoniocaia committed:
Update (Elaborato.java): List<Row> rowsList renamed rows

Co-authored-by: GabrielePuliti <gabriele.puliti@protonmail.com>

antoniocaia committed:
Update (Elaborato.java): method addNewRowToRowsList renamed to addRowToRows

Co-authored-by: GabrielePuliti <gabriele.puliti@protonmail.com>

antoniocaia committed:
Updated (Elaborato.java): Elaborato.java renamed ProcessedFile.java

Updated reference for Elaborato in Parser.java, renamed some variable for consistency.

antoniocaia committed:
Update (Row.java): added links to documentation

Wabri commented on GestioneOrari/src/Elaborato/ProcessedFile.java:
```diff
@@ -0,0 +1,23 @@
+package Elaborato;
```
Magari cambiare il nome del package in inglese tipo "processed"

Wabri commented on GestioneOrari/src/Elaborato/ProcessedFile.java:
```diff
@@ -0,0 +1,23 @@
+		rows = new ArrayList<>();
+	}
+	
+	public void addRowToRows(Date date, String startHour, String endHour, String order, String note) {
```
Non ricordo se nel materiale c'è qualche riferimento al dependency injection, ma in questo caso è chiaro che la richiesta di così tanti parametri è un po' eccessiva per quanto giusta. Qui ci starebbe proprio bene utilizzare quel design pattern. C'è un articolo molto interessante su medium, se vuoi leggerlo lo trovi qui: [What Is Dependency Injection?](https://medium.com/better-programming/what-is-dependency-injection-b2671b1ea90a), che dice:

> "In your code, the interaction between objects should be as thin and clean as possible. You should ask for precisely what you need and nothing more. If you need something, you should ask for it — have it “delivered” to you instead of trying to create it yourself." 

Non so se l'hai mai usato questo design, ma rende tutto più indipendente e pulito.

In questo caso dovresti sostituire gli argomenti del metodo con l'oggetto che vuoi inserire:

```suggestion
	public void addRowToRows(Row row) {
```

Questo design è consigliato nel caso in cui all'interno del metodo chiamato c'è una creazione di un oggetto dipendente dagli argomenti del metodo stesso.

> "Dependency injection is a twenty-five dollar term for a ten-cent concept: If you need something, ask for it. Don’t create things yourself — push that off to someone else. Every class of any substance will likely need help from other classes. If your class needs help, ask for it; don’t try to “do it yourself.” Remember, every time you try to “do it yourself,” you create a hard-coded dependency. And again, those should be avoided like global variables."

@scrivereleonardo @MariusLovesPizza @ebrogini considerate anche voi questo commento che è importante.

antoniocaia replied:
Grazie per la spiegazione, conoscevo il pattern ma non mi sono reso conto fosse applicabile. Ora ho le idee più chiare.

Wabri reviewed (changes requested):
Bravo, qualche altra piccola modifica e ci siamo.

antoniocaia committed:
Update (ProcessedFile): addRowToRows method now takes a Row instance as param
